protocol/README.md

```diff
@@ -8,44 +8,41 @@ The canonical architecture contains four major components.
-  * Any OIDC-compliant IdP software may be used, provided it supports custom claims, and can:
-    * Read the TDF Client public key from a custom HTTP header sent with the OIDC authentication request.
-    * Construct and send an Attribute Provider web service request, including the public key in the payload.
-    * Return the resulting Claims Object in the signed IdP JWT.
-  * A list of Certified OpenID Connect applications can be found at:  https://openid.net/developers/certified/
-  * *OpenTDF Protocol Mapper* (PM) is OpenTDF's Keycloak-specific reference implementation of the above functionality.
-* *Attribute Provider* (AP) - A web service that receives requests which contain information about the authenticated entities from an OIDC IdP with custom claims support (ex: Keycloak with OpenTDF Protocol Mapper), and returns custom TDF OIDC claims in response. It is the responsibility of Attribute Provider to transform incoming 3rd party IdP claims/metadata to a set of outgoing [Attribute Objects](../schema/AttributeObject.md). It returns a TDF [Claims Object](../schema/ClaimsObject.md).
-* *Key Access Service* (KAS) - Responsible for authorizing and granting TDF Clients access to rewrapped data key material. If authorized, TDF Clients (on behalf of themselves, or other entities) can use this rewrapped data key to decrypt TDF ciphertext. A valid OIDC token containing [TDF Claims](../schema/ClaimsObject.md) must be used as a bearer token when communicating with KAS. KAS will verify the authenticity of the bearer token, the request signature, and then the policy claims within that bearer token. An otherwise valid and trusted OIDC token without valid TDF Claims will be rejected.
+* *OpenID Connect (OIDC) Identity Provider (IdP)* - This system could be any OIDC IdP software.Any OIDC-compliant IdP software may be used, provided it supports custom claims and *Demonstration of Proof of Possession (DPoP)*.
+  * Receive a TDF Client public key from a custom HTTP header sent with the OIDC authentication request, or via a side channel or additional request.
+  * Construct and send an Attribute Provider web service request, including sufficient information to identify the client and its capabilities.
+  * In the signed access JWT, include the evidence of the client's public key in the `cnf` claim, and the entitlements in the `tdf_claims`.
+  * A list of Certified OpenID Connect applications can be found at:  https://openid.net/developers/certified/  OpenTDF has chosen Keycloak as its reference implementation IdP.
+    * From Wikipedia:  "Keycloak is an open source software product to allow single sign-on with Identity and Access Management aimed at modern applications and services. As of March 2018 this JBoss community project is under the stewardship of Red Hat.  Keycloak is licensed under Apache 2.0."
+    * *OpenTDF Protocol Mapper* (PM) is OpenTDF's Keycloak-specific reference implementation of the above functionality.
```

Comment:
```suggestion

```

The protocol mapper isn't part of the flow, it is simply a Keycloak specific Java hook to call `entitlement PDP` - it does nothing else and makes no decisions, so it is a low-level implementation detail that doesn't belong in the spec IMO.